Packs/ANYRUN/Integrations/ANYRUN/ANYRUN.py

```diff
@@ -225,32 +409,262 @@ def generate_dbotscore(response):
-            "Type": submission_type,
-            "Vendor": "ANYRUN",
-            "Score": THREAT_TEXT_TO_DBOTSCORE.get(threat_text, 0)
+        main_entity = main_object.get('url')
+        main_entity_type = FeedIndicatorType.URL
+        url_outputs = {
+            'Data': main_object.get('url')
         }
-    }
-    return dbot_score
-
-
-def add_malicious_key(entity, verdict):
+        dbot_score = Common.DBotScore(
+            indicator=main_object.get('url'),
+            indicator_type=DBotScoreType.URL,
+            integration_name='ANYRUN',
+            score=THREAT_TEXT_TO_DBOTSCORE.get(threat_text) or Common.DBotScore.NONE
+        )
+        if dbot_score.score >= 2:
+            url_outputs['Malicious'] = {
+                'Vendor': 'ANYRUN',
+                'Description': threat_text
+            }
+        returned_data.append(CommandResults(
+            outputs_prefix='URL',
+            outputs_key_field=['Data'],
+            outputs=url_outputs,
+            indicator=Common.URL(
+                url=main_object.get('url'),
+                dbot_score=dbot_score,
+            )
+        ))
+
+    # Check if network information is available in the report
+    if 'network' in data:
+        network_data = data.get('network')
+
+        # Then add all the network-related indicators - 'connections'
+        if 'connections' in network_data:
+            connections = network_data.get('connections')
+            for current_connection in connections:
+                reputation = current_connection.get('Reputation')
+                if reputation in reputation_map.keys():
+                    current_dbot_score = Common.DBotScore(
+                        indicator=current_connection.get('IP'),
+                        indicator_type=DBotScoreType.IP,
+                        integration_name='ANYRUN',
+                        score=reputation_map[reputation]
+                    )
+                    relationships = [EntityRelationship(
+                        name=EntityRelationship.Relationships.COMMUNICATED_WITH,
+                        entity_a=main_entity,
+                        entity_a_type=main_entity_type,
+                        entity_b=current_connection.get('IP'),
+                        entity_b_type=FeedIndicatorType.IP,
+                        brand="ANYRUN"
+                    )]
+                    ip_indicator = Common.IP(
+                        ip=current_connection.get('IP'),
+                        asn=current_connection.get('ASN'),
+                        port=current_connection.get('Port'),
+                        geo_country=current_connection.get('Country'),
+                        dbot_score=current_dbot_score,
+                        relationships=relationships
+                    )
+                    if current_connection.get('IP') not in [
+                        x.indicator.ip for x in returned_data if isinstance(x.indicator, Common.IP)
+                    ]:
```

Comment:
Can you elaborate, why if there any IP indicator in the returned_data we wont add another one?
Also, I think we should move this before the creation of the `ip_indicator` because if we don't want to add it to the `CommandsResults` this variable will be redundant.

Reply:
This was becuase there are often multiple connections to the same IP, so it was to save some space in the returned data. But I can remove, as there maybe different port numbers too.